sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/PortableSchemaCoder.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.schemas;
+
+import org.apache.beam.sdk.transforms.SerializableFunctions;
+import org.apache.beam.sdk.values.Row;
+
+/**
+ * A version of SchemaCoder that can only produce/consume Row instances.
+ *
+ * <p>Implements the beam:coders:row:v1 standard coder while still satisfying the requirement that a
+ * PCollection is only considered to have a schema if its coder is an instance of SchemaCoder.
+ */
+public class PortableSchemaCoder extends SchemaCoder<Row> {
```

Comment:
is the reason you need a separate class here because you need a unique Class object for the map?

Reply:
Yeah, I need a unique class for the map in CoderTranslation. I originally tried to use `RowCoder` for this directly, but that didn't work since it needs to be a sub-class of SchemaCoder for PCollection.hasSchema. 

This was the easiest solution I came up with to make it work, but I'm definitely open to other ideas. Maybe I could re-work the CoderTranslation logic so that `SchemaCoder<Row>` instances translate to `beam:coder:row:v1`?

Reply:
I updated this PR to use RowCoder as the Java implementation of `beam:coder:row:v1` and removed `PortableSchemaCoder` now that #9424 is in.